compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -403,24 +413,29 @@ impl<'f> LoopInvariantContext<'f> {
             }
             Binary(binary) => self.can_evaluate_binary_op(binary),
```

Comment:
So it wasn't the `div` being treated as a binary that can be hoisted?

Reply:
This was the issue but the error was occurring earlier inside of `can_hoist_invariant`. `can_be_hoisted` were `hoist_with_predicate == true` needed the extra check that the loop body does not execute, we were only checking that the instruction was not in a control dependent block. `can_evaluate_binary_op` would return `false` for the `div` from the regression as we cannot ensure that the divisor cannot be zero.